src/main/java/mappers/firstvoices/CategoryMapper.java

```diff
@@ -3,63 +3,91 @@
  */
 package mappers.firstvoices;
 
-import mappers.propertyreaders.PropertyReader;
-import org.nuxeo.client.objects.Document;
-
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import mappers.propertyreaders.PropertyReader;
+import org.nuxeo.client.objects.Document;
+import org.nuxeo.client.objects.Documents;
 
 /**
  * @author dyona
  *
  */
-public class CategoryMapper extends RelatedPhraseMapper {
-	
-    @Override
-    protected String getCacheQuery() {
-    	// Include all categories in the Shared Data folder
-        return "SELECT * FROM FVCategory WHERE (fva:dialect IS NULL OR ecm:parentId='" + documents.get("Categories").getId()
-        		+ "') AND ecm:isTrashed = 0 AND ecm:isVersion = 0";
-    }
+public class CategoryMapper extends DictionaryCachedMapper {
 
-    private void updateMainDocumentReference(ArrayList<String> docIDList) {
-        documents.get("current").setPropertyValue(Properties.WORD_CATEGORIES, docIDList);
-    }
+  @Override
+  protected String getCacheQuery() {
 
-    public CategoryMapper() {
-        super("FVCategory", Columns.CATEGORIES);
-        parentKey = "Categories";
-        cacheProperty = Properties.TITLE;
-        propertyReaders.add(new PropertyReader(Properties.WORD_CATEGORIES, Columns.CATEGORIES));
+    Boolean useLocal = isLocalCategories();
+
+    /* For Local Dialect categories add running parameter localCategories */
+    Documents dialectCategoriesDirectory = client.operation("Repository.Query")
+        .param("query", "SELECT * FROM FVCategories "
+            + "WHERE fva:dialect = '" + getDialectID() + "' "
+            + "AND ecm:path STARTSWITH '/FV/Workspaces/' "
+            + "AND dc:title = 'Categories' "
+            + "AND ecm:isTrashed = 0 "
+            + "AND ecm:isVersion = 0")
+        .execute();
+    List<Document> documentsList = dialectCategoriesDirectory.streamEntries().collect(
+        Collectors.toList());
+
+    if (documentsList.size() > 0 && useLocal) {
+      Document categoriesDirectory = documentsList.get(0);
+      String categoriesDirectoryId = categoriesDirectory.getId();
+      return "SELECT * FROM FVCategory WHERE ecm:ancestorId = '" + categoriesDirectoryId
```

Comment:
For good measure, I would add 'ecm:isProxy =0' here (although it should be covered by ecm:ancestorId - it makes it more explicit we want Workspace categories) (RE: FW-1445)